src/test/java/jimmy/mcgymmy/model/food/FoodTest.java

```diff
@@ -7,62 +7,87 @@
 import jimmy.mcgymmy.testutil.FoodBuilder;
 import jimmy.mcgymmy.testutil.TypicalFoods;
 
 
 public class FoodTest {
-    public static final Name VALID_FOOD_NAME = new Name("test food");
-    public static final Name VALID_FOOD_NAME_2 = new Name("test food 2");
     public static final String INVALID_FOOD_NAME = "";
 
-    public static final Protein PROTEIN = new Protein(2);
-    public static final Protein PROTEIN_1 = new Protein(3);
-    public static final Carbohydrate CARBOHYDRATE = new Carbohydrate(3);
-    public static final Carbohydrate CARBOHYDRATE_1 = new Carbohydrate(4);
-    public static final Fat FAT = new Fat(4);
-    public static final Fat FAT_1 = new Fat(5);
+    private static Name validFoodName;
+    private static Name validFoodName2;
+    private static Protein protein;
+    private static Protein protein1;
+    private static Carbohydrate carbohydrate;
+    private static Carbohydrate carbohydrate1;
+    private static Fat fat;
+    private static Fat fat1;
+    private static Food comparedFood;
+    private static Food sameAsComparedFood;
+    private static Food foodWDifferentName;
+    private static Food foodWDifferentProtein;
+    private static Food foodWDifferentCarbs;
+    private static Food foodWDifferentFat;
+
+    static {
+        try {
+            initialiseFoods();
+        } catch (IllegalValueException e) {
+            assert false : "Error in food name";
+        }
+    }
 
-    public static final Food COMPARED_FOOD = new Food(VALID_FOOD_NAME, PROTEIN, FAT, CARBOHYDRATE);
-    public static final Food SAME_AS_COMPARED_FOOD = new Food(VALID_FOOD_NAME, PROTEIN, FAT, CARBOHYDRATE);
-    public static final Food FOOD_W_DIFFERENT_NAME = new Food(VALID_FOOD_NAME_2, PROTEIN, FAT, CARBOHYDRATE);
-    public static final Food FOOD_W_DIFFERENT_PROTEIN = new Food(VALID_FOOD_NAME, PROTEIN_1, FAT, CARBOHYDRATE);
-    public static final Food FOOD_W_DIFFERENT_CARBS = new Food(VALID_FOOD_NAME, PROTEIN, FAT_1, CARBOHYDRATE);
-    public static final Food FOOD_W_DIFFERENT_FAT = new Food(VALID_FOOD_NAME, PROTEIN, FAT, CARBOHYDRATE_1);
+    static void initialiseFoods() throws IllegalValueException {
+        validFoodName = new Name("test food");
+        validFoodName2 = new Name("test food 2");
+        protein = new Protein(2);
+        protein1 = new Protein(3);
+        carbohydrate = new Carbohydrate(3);
+        carbohydrate1 = new Carbohydrate(4);
+        fat = new Fat(4);
+        fat1 = new Fat(5);
+        comparedFood = new Food(validFoodName, protein, fat, carbohydrate);
+        sameAsComparedFood = new Food(validFoodName, protein, fat, carbohydrate);
+        foodWDifferentName = new Food(validFoodName2, protein, fat, carbohydrate);
+        foodWDifferentProtein = new Food(validFoodName, protein1, fat, carbohydrate);
+        foodWDifferentCarbs = new Food(validFoodName, protein, fat1, carbohydrate);
+        foodWDifferentFat = new Food(validFoodName, protein, fat, carbohydrate1);
```

Comment:
These 2 are wrongly labelled. `foodWDifferentCarbs` should use `carbohydrate1` and not `fat1` when initialising and vice verse with `foodWDifferentFat`.